File: lambdas-python/dbQuery.py
# ./lambdas/dbQuery.py

import boto3
import mysql.connector
import json

def lambda_handler(event, context):
    secretsmanager = boto3.client('secretsmanager')
    secret_data = secretsmanager.get_secret_value(SecretId='your-secret-id')
    secret_string = json.loads(secret_data['SecretString'])
    username = secret_string['username']
    password = secret_string['password']
    host = secret_string['host']
    port = secret_string['port']

    print(username)

    connection = mysql.connector.connect(host=host, user=username, password=password, port=port, database='your-database-name')
    cursor = connection.cursor()
    cursor.execute('SELECT * FROM your-table-name')
    rows = cursor.fetchall()

File: lambdas/dbQuery.js
const AWS = require('aws-sdk');
const mysql = require('mysql');

exports.handler = async (event) => {
  const secretsManager = new AWS.SecretsManager();
  const secretData = await secretsManager.getSecretValue({ SecretId: 'freetier/mysql' }).promise();
  const { username, password, host, port } = JSON.parse(secretData.SecretString);

  console.log(username);
  let maskedSecret = password.slice(0, 6) + '*'.repeat(password.length - 10) + password.slice(-4);

  console.log('Masked Secret: ', maskedSecret);

  const connection = mysql.createConnection({
    host: host,
    user: username,
    password: password,
    port: port,
    database: 'database-free-tier'
  });
  connection.connect();
  return new Promise((resolve, reject) => {
    connection.query('SHOW DATABASES', function (error, results, fields) {
        if (error) reject(error);
        else resolve(results);
      });
  });
  
}